h2o-r/h2o-package/R/explain.R

```diff
@@ -2420,29 +2421,63 @@ h2o.ice_plot <- function(model,
           100
         }
       ))
-      tmp[["name"]] <- sprintf("%dth Percentile", i * 10)
+      tmp[["name"]] <- percentile_str
       results <- rbind(results, tmp[, c(column, "name", "mean_response")])
 
       subdata <- as.data.frame(newdata[as.integer(idx),])[[column]]
-        print(is.factor(subdata))
-      column_split = if (is.factor(subdata)) {
-        c(as.character(subdata))
+      if (is.na(subdata)) {
+        # NAs / special values going to be handled in PUBDEV-8493 / NAs in original observations approach should be aligned
+        orig_tmp <- as.data.frame(
+          h2o.partialPlot(
+            models_info$get_model(models_info$model),
+            newdata,
+            column,
+            row_index = as.integer(idx - 1),
+            plot = FALSE,
+            targets = target,
+            include_na = TRUE
+          )
+        )
+        orig_tmp[["name"]] <- percentile_str
+        orig_values <-
+          rbind(orig_values, orig_tmp[nrow(orig_tmp), c(column, "name", "mean_response")])
+        interval <-
+          paste("[", orig_values[nrow(orig_values), c(column)], ", ", orig_values[nrow(orig_values), c("mean_response")], "]", sep =
+                  "")
+        message <-
+          paste(
+            "Original observation of '",
+            column,
+            "' for ",
+            percentile_str,
+            " is ",
+            interval,
+            ". Ploting of NAs is not yet supported.",
+            sep = ""
+          )
+        print(message)
```

Comment:
It's probably better to use `warning(message)` (or `cat(message, "\n")` but I would prefer warning since the user can influence its behavior via `options(warn = ... )`).